civis/tests/test_futures.py

```diff
@@ -30,6 +32,26 @@
     civis_api_spec_channels = json.load(f, object_pairs_hook=OrderedDict)
 
 
+class State:
+    def __init__(self, state):
+        self.state = state
+
+
+@mock.patch(api_import_str, return_value=civis_api_spec_base)
+@mock.patch.object(CivisFuture, '_subscribe')
```

Comment:
Sure. FWIW, I was just trying to mimic the tests as they are now, which use VCR. We can change the whole test suite, but having half of it work one and not the other way is a bit weird.

Reply:
I'm not suggesting we switch away from VCR. The tests in this module aren't using it. What I'm suggesting is that instead of patching the functions which the `APIClient` uses to create itself inside the `CivisFuture`, we create whatever client object we want and pass that explicitly to the `CivisFuture`. The `create_client_mock` is using the same API spec used here.

I see what you're saying about mimicking the pattern used in some of the other tests in this module, but not all of the tests use patching, so the style is already split.